snowflake/data_access_sync_to_target.go

```diff
@@ -24,7 +24,150 @@ import (
 	"github.com/raito-io/cli-plugin-snowflake/common"
 )
 
-func (s *AccessSyncer) generateUniqueExternalId(ap *importer.AccessProvider, prefix string) (string, error) {
+type AccessToTargetSyncer struct {
+	ctx                           context.Context
+	configMap                     *config.ConfigMap
+	namingConstraints             naming_hint.NamingConstraints
+	repo                          dataAccessRepository
+	accessSyncer                  *AccessSyncer
+	accessProviders               *importer.AccessProviderImport
+	accessProviderFeedbackHandler wrappers.AccessProviderFeedbackHandler
+
+	ignoreLinksToRole          []string
+	databaseRoleSupportEnabled bool
+
+	uniqueRoleNameGeneratorsCache map[*string]naming_hint.UniqueGenerator
+	tablesPerSchemaCache          map[string][]TableEntity
+	schemasPerDataBaseCache       map[string][]SchemaEntity
+	warehousesCache               []DbEntity
+}
+
+func NewAccessToTargetSyncer(ctx context.Context, accessSyncer *AccessSyncer, namingConstraints naming_hint.NamingConstraints, repo dataAccessRepository, accessProviders *importer.AccessProviderImport, accessProviderFeedbackHandler wrappers.AccessProviderFeedbackHandler, configMap *config.ConfigMap) *AccessToTargetSyncer {
+	return &AccessToTargetSyncer{
+		accessSyncer:                  accessSyncer,
+		ctx:                           ctx,
+		accessProviders:               accessProviders,
+		accessProviderFeedbackHandler: accessProviderFeedbackHandler,
+		configMap:                     configMap,
+		repo:                          repo,
+		tablesPerSchemaCache:          make(map[string][]TableEntity),
+		schemasPerDataBaseCache:       make(map[string][]SchemaEntity),
+		uniqueRoleNameGeneratorsCache: make(map[*string]naming_hint.UniqueGenerator),
+		namingConstraints:             namingConstraints,
+	}
+}
+
+func (s *AccessToTargetSyncer) syncToTarget() error {
+	defer func() {
+		logger.Info(fmt.Sprintf("Total snowflake query time:  %s", s.repo.TotalQueryTime()))
+		s.repo.Close()
```

Comment:
Same as before. It feels like we should close the repo in the class that created the`AccessToTargetSyncer`